File: packages/tasks-gen/snippets-fixtures/image-to-video/js/huggingface.js/0.fal-ai.js
import { InferenceClient } from "@huggingface/inference";

const client = new InferenceClient(process.env.HF_TOKEN);

const data = fs.readFileSync("cat.png");

const video = await client.imageToVideo({
	provider: "fal-ai",
	model: "Wan-AI/Wan2.2-I2V-A14B",
	inputs: data,
	parameters: { prompt: "The cat starts to dance", },
});

/// Use the generated video (it's a Blob)
// For example, you can save it to a file or display it in a video element

File: packages/tasks-gen/snippets-fixtures/image-to-video/python/huggingface_hub/0.fal-ai.py
from huggingface_hub import InferenceClient, get_token

client = InferenceClient(
    provider="fal-ai",
    api_key=get_token(),
)

with open("cat.png", "rb") as image_file:
   input_image = image_file.read()

video = client.image_to_video(
    input_image,
    prompt="The cat starts to dance",
    model="Wan-AI/Wan2.2-I2V-A14B",
)